packages/agora-rtc-react/src/hooks/tracks.ts

```diff
@@ -127,3 +133,229 @@ export function useVolumeLevel(audioTrack?: IRemoteAudioTrack | ILocalAudioTrack
 
   return volumeLevel;
 }
+
+/**
+ * Auto-subscribe and get remote user audio track.
+ * Unsubscribe track on unmount.
+ */
+export function useRemoteAudioTracks(
+  users: IAgoraRTCRemoteUser[] | undefined,
+  client?: IAgoraRTCClient | null,
+): IRemoteAudioTrack[] {
+  const resolvedClient = useRTCClient(client);
+  const [tracks, setTracks] = useState<IRemoteAudioTrack[]>([]);
+  const isConnected = useIsConnected();
+  const nextTracks = useRef<IRemoteAudioTrack[]>([]);
+
+  useAsyncEffect(async () => {
+    if (!users || !isConnected) return;
+    let isUnmounted = false;
+    const subscribe = async (user: IAgoraRTCRemoteUser) => {
+      try {
+        if (!user.audioTrack && users.some(({ uid }) => user.uid === uid)) {
+          await resolvedClient.subscribe(user, "video").catch(error => {
+            void error;
+          });
+          if (
+            user.audioTrack &&
+            !nextTracks.current.some(track => track.getUserId() === user.uid)
+          ) {
+            nextTracks.current.push(user.audioTrack);
+          }
+
+          nextTracks.current = nextTracks.current.map(track => {
+            if (
+              user.audioTrack &&
+              track.getUserId() === user.uid &&
+              track.getTrackId() !== user.audioTrack.getTrackId()
+            ) {
+              return user.audioTrack;
+            } else {
+              return track;
+            }
+          });
+          if (!isUnmounted) {
+            setTracks(nextTracks.current);
```

Comment:
This `setTracks` is only invoked in `if (!user.audioTrack)`, so this function will do nothing when the `user.audioTrack` exists. -- What if there's no such track in `tracks`?

Reply:
If user-published happens to be triggered when component is unmounted, then it will enter this function. I think it is better to keep it.